lib/index.js

```diff
@@ -3340,6 +3340,51 @@ addMethod('take', function (n) {
     return s;
 });
 
+/**
+ * Creates a new Stream with the first values from the source as long as the function
+ * returns true or a truthy value
+ *
+ * @id takeWhile
+ * @section Transforms
+ * @name Stream.takeWhile(f)
+ * @param {Function} f - the filter function
+ * @throws {TypeError} if `f` is not a function.
+ * @api public
+ *
+ * _([1, 2, 3, 4]).takeWhile(x => x<3) // => 1, 2
+ */
+
+addMethod('takeWhile', function (f) {
+    if (!_.isFunction(f)) {
+        throw new Error('takeWhile expects a function as its only argument.');
+    }
+    return this.consume(function (err, x, push, next) {
+        if (err) {
+            push(err);
+            next();
+        }
+        else if (x === nil) {
+            push(err, x);
+        }
+        else {
+            var fnVal, fnErr;
+            try {
+                fnVal = f(x);
+            }
+            catch (e) {
+                fnErr = e;
+            }
+            if (fnVal) {
```

Comment:
Reverse this.

```
if (fnErr) {
  ...
  next();
} else if (fnVal) {
  ...
  next();
} else {
  push(null, nil);
}
```

You want to check for `fnErr` first, since you need to push `nil` if `fnVal` is `false`.

Reply:
done!